packages/template-retail-react-app/app/components/store-locator-modal/index.jsx

```diff
@@ -16,51 +17,35 @@ import {
     useBreakpointValue
 } from '@salesforce/retail-react-app/app/components/shared/ui'
 import StoreLocatorContent from '@salesforce/retail-react-app/app/components/store-locator-modal/store-locator-content'
-import {useSearchStores} from '@salesforce/commerce-sdk-react'
+
+// Others
 import {
-    DEFAULT_STORE_LOCATOR_COUNTRY_CODE,
+    DEFAULT_STORE_LOCATOR_COUNTRY,
     DEFAULT_STORE_LOCATOR_POSTAL_CODE,
-    STORE_LOCATOR_DISTANCE
+    STORE_LOCATOR_NUM_STORES_PER_LOAD
 } from '@salesforce/retail-react-app/app/constants'
-import {noop} from '@salesforce/retail-react-app/app/utils/utils'
-import {useForm} from 'react-hook-form'
 
-const StoreLocatorModal = ({isOpen, onClose = noop}) => {
-    const intl = useIntl()
+const StoreLocatorModal = ({isOpen, onClose}) => {
+    const [userHasSetManualGeolocation, setUserHasSetManualGeolocation] = useState(false)
 
     const [searchStoresParams, setSearchStoresParams] = useState({
-        countryCode: DEFAULT_STORE_LOCATOR_COUNTRY_CODE,
-        postalCode: DEFAULT_STORE_LOCATOR_POSTAL_CODE
+        countryCode: DEFAULT_STORE_LOCATOR_COUNTRY.countryCode,
+        postalCode: DEFAULT_STORE_LOCATOR_POSTAL_CODE,
+        limit: STORE_LOCATOR_NUM_STORES_PER_LOAD
     })
 
-    const form = useForm({
-        mode: 'onChange',
-        reValidateMode: 'onChange',
-        defaultValues: {
-            countryCode: searchStoresParams.countryCode,
-            postalCode: searchStoresParams.postalCode
-        }
-    })
-    var {data: searchStoresData, isLoading} = useSearchStores({
-        parameters: {
-            countryCode: searchStoresParams.countryCode,
-            postalCode: searchStoresParams.postalCode,
-            locale: intl.locale,
-            maxDistance: STORE_LOCATOR_DISTANCE
-        }
-    })
-    const storesInfo = isLoading ? undefined : searchStoresData?.data || []
+    const isDesktopView = useBreakpointValue({base: false, lg: true})
 
-    const submitForm = async (formData) => {
-        const {postalCode, countryCode} = formData
-        setSearchStoresParams({
-            postalCode: postalCode,
-            countryCode: countryCode
-        })
-    }
+    const storeLocatorContent = (
+        <StoreLocatorContent
+            searchStoresParams={searchStoresParams}
+            setSearchStoresParams={setSearchStoresParams}
+            userHasSetManualGeolocation={userHasSetManualGeolocation}
+            setUserHasSetManualGeolocation={setUserHasSetManualGeolocation}
+        />
+    )
 
-    const isDesktopView = useBreakpointValue({base: false, lg: true})
-    return (
+    return isOpen ? (
```

Comment:
I don't think checking for `isOpen` is needed here because `Modal` is already doing that for us. It will not render its html if the given `isOpen` prop is false.